tests/components/scene/test_init.py

```diff
@@ -222,6 +237,25 @@ async def test_restore_state_does_not_restore_unavailable(
     assert hass.states.get("scene.test").state == STATE_UNKNOWN
 
 
+@pytest.mark.usefixtures("enable_custom_integrations")
+@pytest.mark.parametrize("auto_set", [True, False])
+async def test_set_last_activated(hass: HomeAssistant, auto_set: bool) -> None:
+    """Test the scene with and without auto setting __last_activated."""
+    scene = LastActivatedTestScene(hass, auto_set)
+
+    state = hass.states.get(scene.entity_id)
+    assert state is None
+
+    await scene._async_activate()
+    state = hass.states.get(scene.entity_id)
+
+    if auto_set:
+        assert state is not None
+        assert state.state is not None
+    else:
+        assert state is None
```

Comment:
we should avoid conditions in tests, so lets either have State as part of the test params, or split the test into 2